src/deepsparse/transformers/pipelines_cli.py

```diff
@@ -0,0 +1,269 @@
+        f"Currently supported tasks {SUPPORTED_TASKS.keys()}",
+        choices=SUPPORTED_TASKS.keys(),
+        type=str,
+        default="sentiment-analysis",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--data",
+        help="Path to file containing data for inferences, "
+        f"inputs should be separated via newline, Supports "
+        f"{SUPPORTED_EXTENSIONS} files",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--model-name",
+        "--model_name",
+        help="Canonical name of the hugging face model this model is based on",
+        default=None,
+        type=Optional[str],
+    )
+
+    parser.add_argument(
+        "--model-path",
+        "--model_path",
+        help="Path to model directory containing `model.onnx`, `config.json`,"
+        "and `tokenizer.json` files, ONNX model file, or SparseZoo stub",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--engine-type",
+        "--engine_type",
+        help="Inference engine name to use. Supported options are 'deepsparse'"
+        "and 'onnxruntime'",
+        type=str,
+        choices=SUPPORTED_ENGINES,
+        default=SUPPORTED_ENGINES[0],
+    )
+
+    parser.add_argument(
+        "--config",
+        help="Huggingface model config, if none provided, default will be used"
+        "which will be from the model name or sparsezoo stub if given for "
+        "model path",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--tokenizer",
+        help="Huggingface tokenizer, if none provided, default will be used",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--max-length",
+        "--max_length",
+        help="Maximum sequence length of model inputs. default is 128",
+        type=int,
+        default=128,
+    )
+
+    parser.add_argument(
+        "--num-cores",
+        "--num_cores",
+        help="Number of CPU cores to run engine with. Default is the maximum "
+        "available",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        "--batch_size",
+        help="The batch size to use for pipelines",
+        type=int,
+        default=1,
+    )
+
+    scheduler_choices = ["multi", "single"]
+    parser.add_argument(
+        "--scheduler",
+        choices=scheduler_choices,
+        help="The scheduler to use for the engine. Can be None, single or multi",
+        default=scheduler_choices[0],
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output-file",
+        "--output_file",
+        action="store",
+        dest="output_file",
+        type=str,
+        help="Directs json output to a name of your choice, defaults to " "out.json",
+        default="out.json",
+    )
+
+    _args = parser.parse_args()
+    return _args
+
+
+def _pipeline_inference(
```

Comment:
let's rename this to _process_dataset and move to the `pipelines.py` file as part of our API. also add docstring